### PR TITLE
ci(e2e): upload junit results to gh artifacts

### PR DESCRIPTION
**Pull request by bartsmykla**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

> Changelog: skip

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -138,6 +139,12 @@ jobs:
             target="test/e2e"
           fi
           make ${MAKE_PARAMETERS} CI=true "${target}"
+      - name: "Archive production artifacts"
+        if: ${{ always() }}
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: test-e2e-${{ env.E2E_RUN_NAME }}-${{ (github.event.pull_request.head.sha||github.sha) }}-${{ github.run_id }}
+          path: build/reports/results.xml
```
Comment (slonka) [inline .github/workflows/e2e.yaml:147]: no files found there :/

File: mk/test.mk [hunk 2]
```diff
@@ -7,7 +7,7 @@ GINKGO_UNIT_TEST_FLAGS ?= \
 
 # -race requires CGO_ENABLED=1 https://go.dev/doc/articles/race_detector and https://github.com/golang/go/issues/27089
 UNIT_TEST_ENV=$(GOENV) CGO_ENABLED=1 KUBEBUILDER_ASSETS=$(KUBEBUILDER_ASSETS) TMPDIR=/tmp UPDATE_GOLDEN_FILES=$(UPDATE_GOLDEN_FILES) $(if $(CI),TESTCONTAINERS_RYUK_DISABLED=true,GINKGO_EDITOR_INTEGRATION=true)
-GINKGO_TEST:=$(GINKGO) $(GOFLAGS) $(LD_FLAGS) -p --keep-going --keep-separate-reports --junit-report results.xml --output-dir $(REPORTS_DIR)
+GINKGO_TEST:=$(GINKGO) $(GOFLAGS) $(LD_FLAGS) -p --keep-going --junit-report results.xml --output-dir $(REPORTS_DIR)
```
Comment (slonka) [inline mk/test.mk:10]: keep-separate-reports causes files to be named with test prefix like `build/reports/test_e2e_env_kubernetes_results.xml`. Traced this parameter back to https://github.com/kumahq/kuma/commit/5ee8cb75a88cf33c022e30d2231f98363f8af924#diff-f5975eb03c05409935280a8b9d6538a1d61b21310990e82b24c5121279e23cefR34 - @bartsmykla do you remember why you added it?

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (slonka) [reply .github/workflows/e2e.yaml:147]: https://github.com/kumahq/kuma/actions/runs/8010206149?pr=9354 now it works

File: mk/test.mk (hunk 2 above)
Comment (bartsmykla) [reply mk/test.mk:10]: I unfortunately don't remember

File: .github/workflows/e2e.yaml [hunk 3]
```diff
@@ -21,6 +21,7 @@ env:
   E2E_PARAM_TARGET: ${{ fromJSON(inputs.matrix).target }}
   E2E_PARAM_PARALLELISM: ${{ fromJSON(inputs.matrix).parallelism }}
   CI_TOOLS_DIR: /home/runner/work/kuma/kuma/.ci_tools
+  E2E_RUN_NAME: ${{ fromJSON(inputs.matrix).target }}-${{ fromJSON(inputs.matrix).k8sVersion }}-${{ fromJSON(inputs.matrix).cniNetworkPlugin }}-${{ fromJSON(inputs.matrix).arch }}-${{ fromJSON(inputs.matrix).parallelism }}
```
Comment (slonka) [inline .github/workflows/e2e.yaml:24]:
```suggestion
  E2E_RUN_NAME: ${{ fromJSON(inputs.matrix).target }}_${{ fromJSON(inputs.matrix).k8sVersion }}_${{ fromJSON(inputs.matrix).cniNetworkPlugin }}_${{ fromJSON(inputs.matrix).arch }}_${{ fromJSON(inputs.matrix).parallelism }}
```

File: .github/workflows/e2e.yaml [hunk 4]
```diff
@@ -138,6 +139,14 @@ jobs:
             target="test/e2e"
           fi
           make ${MAKE_PARAMETERS} CI=true "${target}"
+      - name: "Archive production artifacts"
+        if: ${{ always() }}
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: test-e2e-${{ env.E2E_RUN_NAME }}-${{ (github.event.pull_request.head.sha||github.sha) }}-${{ github.run_id }}
```
Comment (slonka) [inline .github/workflows/e2e.yaml:146]:
```suggestion
          name: test_e2e_${{ env.E2E_RUN_NAME }}-${{ (github.event.pull_request.head.sha||github.sha) }}-${{ github.run_id }}
```

to comform with op framework requirements

File: .github/workflows/e2e.yaml [hunk 5]
```diff
@@ -21,6 +21,7 @@ env:
   E2E_PARAM_TARGET: ${{ fromJSON(inputs.matrix).target }}
   E2E_PARAM_PARALLELISM: ${{ fromJSON(inputs.matrix).parallelism }}
   CI_TOOLS_DIR: /home/runner/work/kuma/kuma/.ci_tools
+  E2E_RUN_NAME: ${{ fromJSON(inputs.matrix).target }}_${{ fromJSON(inputs.matrix).k8sVersion }}_${{ fromJSON(inputs.matrix).cniNetworkPlugin }}_${{ fromJSON(inputs.matrix).arch }}_${{ fromJSON(inputs.matrix).parallelism }}
```
Comment (lahabana) [inline .github/workflows/e2e.yaml:24]: Hmm this is not very future proof with actions in km. There's likely something better with `inputs`, the `context` or something like this no?

**Pull request merged**